packages/catalog-process/src/services/validators.ts

```diff
@@ -274,16 +275,29 @@ export async function assertNotDuplicatedEServiceNameForProducer(
   producerId: TenantId,
   readModelService: ReadModelService
 ): Promise<void> {
-  const eserviceWithSameName =
-    await readModelService.getEServiceByNameAndProducerId({
+  const eserviceWithSameNameExists =
+    await readModelService.existsEServiceWithNameAndProducerId({
       name,
       producerId,
     });
-  if (eserviceWithSameName !== undefined) {
+  if (eserviceWithSameNameExists) {
     throw eServiceNameDuplicateForProducer(name, producerId);
   }
 }
 
+export async function assertNoEServiceTemplateNameConflict(
```

Comment:
See this comment left on the PR for eservice template names: https://github.com/pagopa/interop-be-monorepo/pull/1970/files#r2135781113

If you address it there, align also this